pkg/security/secl/rules/ruleset.go

```diff
@@ -524,9 +524,10 @@ func (rs *RuleSet) runRuleActions(ctx *eval.Context, rule *Rule) error {
 }
 
 // Evaluate the specified event against the set of rules
-func (rs *RuleSet) Evaluate(event eval.Event) bool {
+func (rs *RuleSet) Evaluate(event eval.Event, SecurityProfileAutoSuppression bool) bool {
```

Comment:
Happy to chat about it, but I think we should try to isolate the ruleset a bit more from security profile: we could, for example, conditionally call the `Evaluate` depending on the event flags (just like what we do for AD, see [here](https://github.com/DataDog/datadog-agent/blob/a1f863d998596d0ce5f708eb2c67a4c183ac3b10/pkg/security/module/cws.go#L411)).

If it did match, and if we need to push a discarder, I'd add a `pushDiscarder` boolean input to the `RuleSet` function 🤔

Reply:
Agree on this, the rule evaluation should be isolated as the rule check use this and is used by other tool.